src/app/article-list-data.service.ts

```diff
@@ -1,17 +1,36 @@
 import { Injectable } from "@angular/core";
 import { Article } from "./article";
-import content from "../assets/content.json";
+import { Observable } from "rxjs";
+import { map } from "rxjs/operators";
+import { FetchArticlesService } from "./fetch-articles.service";
 
 @Injectable({
   providedIn: "root"
 })
 export class ArticleListDataService {
   public articleListData: Article[];
-  constructor() {
-    this.articleListData = content;
+  public articleList: Observable<Article[]>;
+  constructor(private fetchArticle: FetchArticlesService) {
+    this.articleList = fetchArticle.fetchArticleList();
   }
-  getArticleList(): Article[] {
-    console.log(this.articleListData);
-    return this.articleListData;
+
+  getArticleList(): Observable<Article[]> {
+    return this.articleList.pipe(
+      map(responsedata => {
```

Comment:
Here the use of the `map` operator can be removed since it isn't transforming the `responseData`, it is just returning it. So this method implementation can be changed to 
```
getArticleList(): Observable<Article[]> {
    return this.articleList$;
}
```

Reply:
`map` operator removed